### ISSUE TITLE
Type check with mypy

### ISSUE THREAD

**Issue by jonafato** [kjaymiller/Python-Community-News#53]
We have some code that's type hinted, so let's make full use of it.

**Comment by kjaymiller** [comment]
resolved in #109 and updated to be included in pre-commit in #110

**Comment by jonafato** [comment]
I don't think this is resolved by #109; that fixes the typing issues at the moment but doesn't enforce anything. I think #110 is a duplicate of this issue.
